### PR TITLE
[BugFix] [Kernel] Fix GPU SEGV occuring in fused_moe kernel

### PR DESCRIPTION
**Pull request by rasmith**
When running a large model (~500G), I encountered GPU SEGV.   It was occurring in `fused_moe_kernel`:

```
Thread 98 "python" received signal SIGSEGV, Segmentation fault.
Warning: precise memory violation signal reporting is not enabled, reported
location may not be accurate.  See "show amdgpu precise-memory".
[Switching to thread 98, lane 0 (AMDGPU Lane 7:2:1:1/0 (0,0,0)[0,0,0])]
fused_moe_kernel () at fused_moe.py:142
142             b = tl.load(b_ptrs,
```

 It seemed to me that this could be integer overflow.  Changing offsets to use `tl.int64` in `fused_moe.py:fused_moe_kernel `fixes the issue.

### CODE REVIEW

**Review by tlrmchlsmth — COMMENTED** [review]
Thanks for the fix! I noted a couple of spots where I think 32-bits should be fine, WDYT?

File: vllm/model_executor/layers/fused_moe/fused_moe.py [hunk 1]
```diff
@@ -105,16 +105,18 @@ def fused_moe_kernel(
     num_tokens_post_padded = tl.load(num_tokens_post_padded_ptr)
     if pid_m * BLOCK_SIZE_M >= num_tokens_post_padded:
         return
-    offs_token_id = pid_m * BLOCK_SIZE_M + tl.arange(0, BLOCK_SIZE_M)
-    offs_token = tl.load(sorted_token_ids_ptr + offs_token_id)
+    offs_token_id = pid_m * BLOCK_SIZE_M + tl.arange(0, BLOCK_SIZE_M).to(
+        tl.int64)
+    offs_token = tl.load(sorted_token_ids_ptr + offs_token_id).to(tl.int64)
```
[the remark this answers is not in the document]
Comment (rasmith) [reply vllm/model_executor/layers/fused_moe/fused_moe.py]: Changed back and still works.

File: vllm/model_executor/layers/fused_moe/fused_moe.py [hunk 2]
```diff
@@ -105,16 +105,18 @@ def fused_moe_kernel(
...
     token_mask = offs_token < num_valid_tokens
 
-    offs_bn = (pid_n * BLOCK_SIZE_N + tl.arange(0, BLOCK_SIZE_N)) % N
-    offs_k = tl.arange(0, BLOCK_SIZE_K)
+    offs_bn = (pid_n * BLOCK_SIZE_N +
+               tl.arange(0, BLOCK_SIZE_N).to(tl.int64)) % N
+    offs_k = tl.arange(0, BLOCK_SIZE_K).to(tl.int64)
```
[the remark this answers is not in the document]
Comment (rasmith) [reply vllm/model_executor/layers/fused_moe/fused_moe.py]: Changed back and still works for second one and still works.

File: vllm/model_executor/layers/fused_moe/fused_moe.py [hunk 3]
```diff
@@ -105,16 +105,18 @@ def fused_moe_kernel(
...
     a_ptrs = a_ptr + (offs_token[:, None] // top_k * stride_am +
                       offs_k[None, :] * stride_ak)
 
-    off_experts = tl.load(expert_ids_ptr + pid_m)
+    off_experts = tl.load(expert_ids_ptr + pid_m).to(tl.int64)
```
[the remark this answers is not in the document]
Comment (rasmith) [reply vllm/model_executor/layers/fused_moe/fused_moe.py:119]: Changed back and still works.

[the remark this answers is not in the document]
Comment (tlrmchlsmth) [reply vllm/model_executor/layers/fused_moe/fused_moe.py:119]: Did you forget this one?

[the remark this answers is not in the document]
Comment (rasmith) [reply vllm/model_executor/layers/fused_moe/fused_moe.py]: OK, actually, I tried removing this one, and I still get SEGV.

[the remark this answers is not in the document]
Comment (rasmith) [reply vllm/model_executor/layers/fused_moe/fused_moe.py]: I thought I had change it before, but missed it, but when I actually try to change it back to 32-bits it still gives SEGV.

**Review by tlrmchlsmth — APPROVED** [review]

File: vllm/model_executor/layers/fused_moe/fused_moe.py [hunk 4]
```diff
@@ -116,7 +116,7 @@ def fused_moe_kernel(
     a_ptrs = a_ptr + (offs_token[:, None] // top_k * stride_am +
                       offs_k[None, :] * stride_ak)
 
-    off_experts = tl.load(expert_ids_ptr + pid_m).to(tl.int64)
+    off_experts = tl.load(expert_ids_ptr + pid_m)
```
Comment (rasmith) [inline vllm/model_executor/layers/fused_moe/fused_moe.py:119]: @tlrmchlsmth  Doing this causes SEGV when I tested.

**Comment by rasmith** [comment]
DCO issues, need to close and reopen.

**Pull request closed without merging**